streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamTask.java

```diff
@@ -278,6 +281,10 @@ public void initializeIfNeeded() {
         }
     }
 
+    public void setConsumedOffsetsAndMetadata(final Map<TopicPartition, OffsetAndMetadata> consumedOffsetsAndMetadata) {
+        this.consumedOffsetsAndMetadata = consumedOffsetsAndMetadata;
```

Comment:
I think these are not the already consumed offsets and metadata but the next offsets to be consumed, right? I would reflect this in the naming.

Maybe `nextOffsetsAndMetadataToBeConsumed`, even if lengthy.